extensions/core/runtime/src/main/java/org/apache/camel/quarkus/core/CamelConfig.java

```diff
@@ -174,4 +180,48 @@
         public Optional<List<String>> includePatterns;
     }
 
+    @ConfigGroup
+    public static class RuntimeCatalogConfig {
+        /**
+         * Enable {@link CamelRuntimeCatalog} functionaries.
```

Comment:
I have just learned about the existence of `functionary/functionaries` as a valid work of the English language. Anyway: is that not a typo?

In the description of the option, I'd like to learn what exactly happens when the option is true and what happens otherwise. 

If I am a new user, I may have no idea what is CamelRuntimeCatalog. Yes there is a link to CamelRuntimeCatalog, but that class has no JavaDoc. Moreover, the link will most probably not work everywhere (IDEs, Web docs) where the text will be displayed. So either a working link to a resource explaining the concept of the catalog or a short inline definition would be nice.

It would also be nice to document the relationship (if there is any) to other options in this file - e.g. does enabled = false imply that all other options in this file are ignored?

Does this option only matter for the native mode? What happens in the JVM mode then?

Reply:
> I have just learned about the existence of `functionary/functionaries` as a valid work of the English language. Anyway: is that not a typo?
> 
> In the description of the option, I'd like to learn what exactly happens when the option is true and what happens otherwise.
> 

It is a typo

> If I am a new user, I may have no idea what is CamelRuntimeCatalog. Yes there is a link to CamelRuntimeCatalog, but that class has no JavaDoc. Moreover, the link will most probably not work everywhere (IDEs, Web docs) where the text will be displayed. So either a working link to a resource explaining the concept of the catalog or a short inline definition would be nice.
> 

I don't think we need to describe what the RuntimeCamelContext is in camel-quarkus so if there's lack of documentation on camel side, please raise an issue there. 

> It would also be nice to document the relationship (if there is any) to other options in this file - e.g. does enabled = false imply that all other options in this file are ignored?
> 
> Does this option only matter for the native mode? What happens in the JVM mode then?

Agree, this need to be improved

Reply:
should be fixed now